modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/dsl/StatsD.scala

```diff
@@ -0,0 +1,92 @@
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.rdbloader.dsl
+
+import java.time.Duration
+import java.net.{DatagramPacket, DatagramSocket, InetAddress}
+import java.nio.charset.StandardCharsets.UTF_8
+
+import cats.Monad
+import cats.implicits._
+
+import cats.effect.{Blocker, ContextShift, Resource, Sync, Timer}
+
+import com.snowplowanalytics.snowplow.rdbloader.common.LoaderMessage
+import com.snowplowanalytics.snowplow.rdbloader.common.config.Config
+
+trait StatsD[F[_]] {
+  def sendMetrics(loaded: LoaderMessage.ShreddingComplete): F[List[StatsD.KVMetric]]
+}
+
+object StatsD {
+
+  final case class KVMetric(key: String, value: String)
+
+  private val MinCollectorLatencyName  = "latency.collector.min"
+  private val MaxCollectorLatencyName  = "latency.collector.max"
+  private val LoaderLatencyName        = "latency.loader"
+  private val CountGoodName            = "count.good"
+
+  def getInstance[F[_]: ContextShift: Monad: Sync: Timer](statsDConfig: Option[Config.StatsD], blocker: Blocker): StatsD[F] =
+    statsDConfig match {
+      case Some(config) =>
+        new StatsD[F] {
+          def sendMetrics(loaded: LoaderMessage.ShreddingComplete): F[List[KVMetric]] =
+            mkSocket[F](blocker).use { socket =>
+              for {
+                metrics <- getMetrics[F](loaded)
+                formatted = metrics.map(statsDFormat(config))
+                ip <- blocker.delay(InetAddress.getByName(config.hostname))
+                _ <- formatted.traverse_(sendMetric[F](blocker, socket, ip, config.port))
+              } yield (metrics)
+           }
+        }
+      case None =>
+        new StatsD[F] {
+          def sendMetrics(loaded: LoaderMessage.ShreddingComplete): F[List[KVMetric]] = getMetrics[F](loaded)
+        }
+    }
+
+  private def mkSocket[F[_]: ContextShift: Sync](blocker: Blocker): Resource[F, DatagramSocket] =
+    Resource.fromAutoCloseableBlocking(blocker)(Sync[F].delay(new DatagramSocket))
+
+  private def getMetrics[F[_]: Monad: Timer](loaded: LoaderMessage.ShreddingComplete): F[List[KVMetric]] =
+    Timer[F].clock.instantNow.map { now =>
+      List(
+        loaded.timestamps.min.map(min => Duration.between(min, now).toSeconds()).map(l => KVMetric(MaxCollectorLatencyName, l.toString)),
+        loaded.timestamps.max.map(max => Duration.between(max, now).toSeconds()).map(l => KVMetric(MinCollectorLatencyName, l.toString)),
+        Some(Duration.between(loaded.timestamps.jobCompleted, now).toSeconds()).map(l => KVMetric(LoaderLatencyName, l.toString)),
+        loaded.count.map(c => KVMetric(CountGoodName, c.toString))
+      )
+        .collect { case Some(m) => m }
```

Comment:
```suggestion
      ).unite
```

`unite` comes from [cats.Alternative](https://github.com/typelevel/cats/blob/main/core/src/main/scala/cats/Alternative.scala#L24-L27) and exists exactly for `List[Option[A]]` casts. `flatten` would also do the trick, but I don't like it because it implies `F[F[A]]` and works just because of Scala "hacks"

Reply:
That's great :ok_hand: